lib/driver.js

```diff
@@ -411,11 +416,22 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.quit();
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
-    } else if (!util.hasValue(this.wda.webDriverAgentUrl) && (await this.wda.isRunning())) {
-      log.info(`Will reuse previously cached WDA instance at '${this.wda.url.href}'. ` +
-               `Set the wdaLocalPort capability to a value different from ${this.wda.url.port} ` +
-               `if this is an undesired behavior.`);
-      this.wda.webDriverAgentUrl = this.wda.url.href;
+    } else if (!util.hasValue(this.wda.webDriverAgentUrl)) {
+      const currentAppBundleID = await this.wda.getCurrentBundleId;
+      if (util.hasValue(currentAppBundleID)) {
+        if (util.hasValue(this.opts.updatedWDABundleId) && (this.opts.updatedWDABundleId !== currentAppBundleID)) {
+          log.info(`Will stop running WDA since the WDA has different bundle id. The WDA is '${currentAppBundleID}'.`);
+          // TODO: this.wda.quit() just stops iproxy, xcodebuild  and jwproxy. Not WDA process which is already running.
```

Comment:
so why we don't uninstall WDA then?

Reply:
yeah, I thought so. It was my note before sleeping 😅